python/nano/src/bigdl/nano/dispatcher.py

```diff
@@ -0,0 +1,130 @@
+        ]
+
+    if patch_torch and _mapping_torch is None:
+        _mapping_torch = []
+        import pytorch_lightning
+        import torchvision
+        from bigdl.nano.pytorch import Trainer
+        from bigdl.nano.pytorch.vision import transforms
+        from bigdl.nano.pytorch.vision import datasets
+        _mapping_torch += [
+            [pytorch_lightning, "Trainer", Trainer, None],
+            [torchvision, "transforms", transforms, None],
+            [torchvision, "datasets", datasets, None],
+        ]
+
+    if not patch_tf:
+        return [], _mapping_torch
+    if not patch_torch:
+        return _mapping_tf, []
+    return _mapping_tf, _mapping_torch
+
+
+def patch_nano(patch_tf=None, patch_torch=None):
+    """
+    patch_nano is used to patch optimized class to replace original ones.
+
+    Optimized classes include:
+
+    | 1. tf.keras.Model/keras.Model -> bigdl.nano.tf.keras.Model
+    | 2. tf.keras.Sequential/keras.Sequential -> bigdl.nano.tf.keras.Sequential
+    | 3. tf.keras.layers.Embedding/keras.layers.Embedding -> bigdl.nano.tf.keras.layers.Embedding
+    | 4. tf.optimizers.Adam -> bigdl.nano.tf.optimizers.SparseAdam
+    | 5. pytorch_lightning.Trainer -> bigdl.nano.pytorch.Trainer
+    | 6. torchvision.transforms -> bigdl.nano.pytorch.vision.transforms
+    | 7. torchvision.datasets -> bigdl.nano.pytorch.vision.datasets
+
+    :param patch_tf: bool, if patch tensorflow related classes, will patch defaultly if tensorflow
+           is installed
+    :param patch_torch: bool, if patch pytorch related classes, will patch defaultly if pytorch
+           is installed
+    """
+    if patch_tf is None:
+        patch_tf = find_spec("tensorflow") is not None
+    if patch_torch is None:
+        patch_torch = find_spec("pytorch_lightning") is not None
+
+    mapping_tf, mapping_torch = _get_patch_map()
+
+    if patch_tf:
+        for mapping_iter in mapping_tf:
+            mapping_iter[3] = getattr(mapping_iter[0], mapping_iter[1], None)
+            setattr(mapping_iter[0], mapping_iter[1], mapping_iter[2])
+
+    if patch_torch:
+        for mapping_iter in mapping_torch:
+            mapping_iter[3] = getattr(mapping_iter[0], mapping_iter[1], None)
+            setattr(mapping_iter[0], mapping_iter[1], mapping_iter[2])
+
+
+def unpatch_nano(unpatch_tf=None, unpatch_torch=None):
+    """
+    unpatch_nano is used to unpatch optimized class to original ones.
+
+    :param unpatch_tf: bool, if unpatch tensorflow related classes,
+           will unpatch defaultly if tensorflow is installed
+    :param unpatch_torch: bool, if unpatch pytorch related classes,
+           will unpatch defaultly if pytorch is installed
+    """
+    if unpatch_tf is None:
+        unpatch_tf = find_spec("tensorflow") is not None
+    if unpatch_torch is None:
+        unpatch_torch = find_spec("pytorch_lightning") is not None
+
+    mapping_tf, mapping_torch = _get_patch_map()
```

Comment:
maybe just directly use global variable `_mapping_tf ` and `_mapping_torch `?

Say in jupyter notebook:
1. user installs tensorflow
2. user calls patch nano
3. user decide to try pytorch and install pytorch
4. user calls unpatch nano

Is the behavior correct in this scenario?

Reply:
will check and give feedbacks here.

Reply:
well it won't happen under current API design. But I tried this:
1. users install pytorch-lightning
2. users call patch torch
3. users uninstall pytorch-lightning
4. users call unpatch torch
Although I don't think this is a common use-case, but at least the api calling does not cause errors.